### PR TITLE
Add grin package

### PR DESCRIPTION
**Pull request by sjagoe**
Package https://pypi.python.org/pypi/grin, a grep-like program in Python with sane defaults for searching code.

### CODE REVIEW

**Comment by iElectric** [comment]
if it's a cli tool, put it in `all-packages.nix`

**Comment by sjagoe** [comment]
I have moved the call to `all-packages.nix` and the derivation to `tools/text/grin/default.nix` (based on where `gnugrep` is located). Is that the right thing to do?

File: pkgs/tools/text/grin/default.nix [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    url = "https://pypi.python.org/packages/source/g/grin/${name}.tar.gz";
+    sha256 = "1swzwb17wibam8jszdv98h557hlx44pg6psv6rjz7i33qlxk0fdz";
+  };
+
+  buildInputs = with pythonPackages; [ nose ];
+  propagatedBuildInputs = with pythonPackages; [ argparse ];
+
+  meta = {
+    homepage = https://pypi.python.org/pypi/grin;
+    description = "A grep program configured the way I like it.";
```
Comment (iElectric) [inline pkgs/tools/text/grin/default.nix]: set `meta.platform` and `meta.maintainers`

**Comment by iElectric** [comment]
Yup, just one small note :)

File: pkgs/tools/text/grin/default.nix (hunk 1 above)
Comment (sjagoe) [inline pkgs/tools/text/grin/default.nix]: I guess I am the maintainer, as the author won't be using nixos. I guess I add myself to `lib/maintainers.nix`.

**Pull request merged**